pandas/tests/io/test_gcs.py

```diff
@@ -1,115 +0,0 @@
-from io import StringIO
```

Comment:
I think we still wantthis test file.

Reply:
The GCS and S3 tests were not really testing either of those APIs, they both had (different) FS mocks which were a lot like fsspec's memory FS.

Reply:
Do the tests pass with the new implementation, or are the mocks not valid?

At the least, we'll want replacements for each of these tests.

Reply:
That's what test_fsspec.py  does. The existing tests in the two files were essentially the same with mock FSs (i.e., didn't actually touch the gcsfs or s3fs code at all).
Actually, I notice that moto is listed in several of the test envs, so we should be able to build out a real suite for s3 and maybe a couple of select fsspec protocols which don't have dependencies (file, memory and ... ?).